coap-core/src/main/java/com/mbed/coap/server/block/BlockWiseIncomingFilter.java

```diff
@@ -144,9 +145,12 @@ private CoapResponse updateBlockResponse(final BlockOption block2Response, final
         if (blTo + 1 >= resp.getPayload().size()) {
             blTo = resp.getPayload().size();
             block2Res = new BlockOption(block2Res.getNr(), block2Res.getBlockSize(), false);
-        } else {
+        } else if (!resp.getCode().isError()) {
             block2Res = new BlockOption(block2Res.getNr(), block2Res.getBlockSize(), true);
+        } else {
+            throw new RuntimeErrorException(new Error(), "Too big payload for error response.");
```

Comment:
What if we simply check if response is an error and then return it without modifying it. It could be done in `adjustPayloadSize` function, something like:
```
        if (resp.getCode().isError()) {
            return resp;
        }
```

Reply:
Not sure. It would allow sending bigger payload than client requested. But then again, the one implementing the server can decide to send smaller error.